Python/Include/Arithmetics.hpp

```diff
@@ -0,0 +1,111 @@
+#ifndef ROVER_PYTHON_ARITHMETICS_HPP
+#define ROVER_PYTHON_ARITHMETICS_HPP
+#include <type_traits>
+#include <pybind11/pybind11.h>
+
+namespace Rover::Details {
+  template<typename... Args>
+  struct contain_objects {
+    static constexpr bool value = std::disjunction_v<std::is_convertible<
+      Args, pybind11::object>...>;
+  };
+
+  template<typename... Args>
+  inline constexpr bool contain_objects_v = contain_objects<Args...>::value;
+
+  template<typename T>
+  auto ensure_object(const T& value) {
+    if constexpr(contain_objects_v<T>) {
+      return value;
+    } else {
+      return pybind11::cast(value);
+    }
+  }
+
+  const pybind11::module& import_operator();
+}
+
+namespace pybind11 {
+  object abs(const object& obj);
+  object floor(const object& obj);
+
+  template<typename T1, typename T2, std::enable_if_t<
+    Rover::Details::contain_objects_v<T1, T2>>* = nullptr>
+  bool operator <(const T1& lhs, const T2& rhs) {
```

Comment:
Prefer to use ```enable_if``` on the return type to avoid making the enable_if intrude on the API. The only time when this can't be done is when using ```enable_if``` on a constructor in which case you have to intrude on the API either through a dummy template parameter or dummy constructor parameter.